code/obj/flock/structure/annihilator.dm

```diff
@@ -0,0 +1,144 @@
+	var/checker_radius = 2
+	/// list of projectile checker objects the annihilator uses to work
+	var/list/projectile_checkers = null
+
+	New(atom/location, datum/flock/F = null)
+		..()
+		var/turf/src_turf = get_turf(src)
+		var/list/turfs = block(locate(max(src_turf.x - src.checker_radius, 1), max(src_turf.y - src.checker_radius, 1), src_turf.z), locate(min(src_turf.x + src.checker_radius, world.maxx), min(src_turf.y + src.checker_radius, world.maxy), src_turf.z))
+
+		src.projectile_checkers = list()
+		for (var/turf/T as anything in turfs)
+			src.projectile_checkers += new /obj/annihilator_projectile_checker(T, src)
+
+		src.info_tag.set_info_tag("Not generating bolt")
+
+	building_specific_info()
+		return src.check_bolt_status() + "."
+
+	process(mult)
+		if (src.flock?.can_afford_compute(src.online_compute_cost))
+			src.compute = -src.online_compute_cost
+			if (!src.powered)
+				ON_COOLDOWN(src, "bolt_gen_time", 10 SECONDS)
+				src.update_flock_compute("apply")
+				src.powered = TRUE
+		else if (!src.flock || src.flock.used_compute > src.flock.total_compute() || !src.powered)
+			if (src.powered)
+				if (src.flock)
+					src.update_flock_compute("remove")
+				if (src.checkers_powered)
+					src.power_projectile_checkers(FALSE)
+			src.compute = 0
+			src.powered = FALSE
+
+		if (src.powered)
+			if (GET_COOLDOWN(src, "bolt_gen_time"))
+				src.icon_state = "annihilator-generating"
+				if (src.checkers_powered)
+					src.power_projectile_checkers(FALSE)
+			else
+				src.icon_state = "annihilator-ready"
+				if (!src.checkers_powered)
+					src.power_projectile_checkers(TRUE)
+		else
+			src.icon_state = "annihilator-off"
+
+		src.info_tag.set_info_tag(src.check_bolt_status())
+
+	proc/power_projectile_checkers(state)
+		for (var/obj/annihilator_projectile_checker/checker as anything in src.projectile_checkers)
+			checker.on = state
+		src.checkers_powered = state
+
+	proc/check_bolt_status()
+		if (!src.powered)
+			return "Not generating bolt"
+		else if (GET_COOLDOWN(src, "bolt_gen_time"))
+			return "Generation time left: [round(GET_COOLDOWN(src, "bolt_gen_time") / 10)] seconds"
+		else
+			return "Bolt ready"
+
+	proc/activate(obj/projectile/bullet)
+		ON_COOLDOWN(src, "bolt_gen_time", 10 SECONDS)
+		src.icon_state = "annihilator-generating"
+		src.power_projectile_checkers(FALSE)
+		var/list/gnesis_bolt_objs = DrawLine(src, bullet, /obj/line_obj/gnesis_bolt, 'icons/obj/projectiles.dmi', "WholeGnesisBolt", TRUE, TRUE, "HalfStartGnesisBolt", "HalfEndGnesisBolt")
+		SPAWN(0.25 SECONDS)
+			for (var/obj/line_obj/gnesis_bolt/gnesis_bolt_obj as anything in gnesis_bolt_objs)
+				qdel(gnesis_bolt_obj)
+		playsound(src, 'sound/weapons/railgun.ogg', 50, TRUE) // placeholder sound
+		qdel(bullet)
+
+	disposing()
+		for (var/obj/annihilator_projectile_checker/checker as anything in src.projectile_checkers)
+			qdel(checker)
+		..()
+
+
+/obj/annihilator_projectile_checker
+	name = null
+	desc = null
+	anchored = TRUE
```

Comment:
Should this use anchored 2?

Reply:
Uh, I didn't realize that existed. All Flock structures use anchored = TRUE, so I think this will probably break somewhat near a black hole if anchored = 2, less so if anchored is 1